File: doris-3/sangbok/submissions/accepted/sangbok.py
#!/usr/bin/env python3

def solve(t, s):
    s_max = 60*t

    s.sort()
    sum = 0

    for e in s:
        sum += e

        if sum > s_max:
            sum -= e
            break

    return sum



t, n = map(int, input().split())
s = list(map(int, input().split()))
sum = solve(t, s)

print(sum)

File: doris-3/sangbok/submissions/accepted/sangbok.rs
use std::io::stdin;
use std::collections::BinaryHeap;

fn main() {
    let mut string = String::new();
    stdin().read_line(&mut string).unwrap();

    let mut iter = string.trim().split(" ").map(|x| x.parse::<u64>().unwrap());
    let s_max = iter.next().unwrap()*60;
    let _n = iter.next().unwrap();

    string.clear();
    stdin().read_line(&mut string).unwrap();

    let mut sum = 0;
    let mut queue = BinaryHeap::new();

    for s in string.trim().split(" ").map(|x| x.parse::<u64>().unwrap()) {
        queue.push(s);
        sum += s;

        if sum > s_max {
            sum -= queue.pop().unwrap();
        }
    }

    println!("{}", sum);
}
